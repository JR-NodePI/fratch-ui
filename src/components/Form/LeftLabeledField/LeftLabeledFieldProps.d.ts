export type LeftLabeledFieldProps = {
  className?: string;
  field: JSX.Element;
  label: JSX.Element;
};
