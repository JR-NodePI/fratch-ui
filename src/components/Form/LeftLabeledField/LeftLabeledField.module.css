.wrapper {
  display: flex;
  flex-wrap: nowrap;
  align-items: stretch;
  position: relative;
  margin-bottom: 8px;
}
.wrapper .label_container {
  display: block;
  background-color: var(--ft-color-grey-3);
  border-left: var(--ft-form-item-border);
  border-top: var(--ft-form-item-border);
  border-bottom: var(--ft-form-item-border);
  padding: 0 19px 0 16px;
  margin-right: -3px;
  white-space: nowrap;
  border-radius: var(--ft-form-item-border-radius) 0 0
    var(--ft-form-item-border-radius);
  box-shadow: var(--ft-box-shadow-1);
  position: relative;
  z-index: 1;
  align-items: center;
  display: flex;
}
.wrapper .field_container {
  min-width: 200px;
  width: 100%;
  z-index: 2;
}
