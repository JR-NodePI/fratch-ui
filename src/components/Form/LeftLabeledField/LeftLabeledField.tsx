import { c } from '../../../helpers/classNameHelpers';

import styles from './LeftLabeledField.module.css';

function LeftLabeledField({
  className,
  field,
  label,
}: {
  className?: string;
  field: JSX.Element;
  label: JSX.Element;
}): JSX.Element {
  return (
    <div className={c(styles.wrapper, className)}>
      <div className={c(styles.label_container)}>{label}</div>
      <div className={c(styles.field_container)}>{field}</div>
    </div>
  );
}

export default LeftLabeledField;
