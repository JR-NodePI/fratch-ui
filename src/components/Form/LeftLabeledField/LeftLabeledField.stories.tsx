import type { Meta, StoryObj } from '@storybook/react';

import Select from '../Select/Select';
import { SelectOption } from '../Select/SelectProps';
import LeftLabeledField from './LeftLabeledField';

const meta = {
  title: 'Example/Form/LeftLabeledField',
  component: LeftLabeledField,
  tags: ['autodocs'],
  argTypes: {
    field: { control: false },
    label: { control: false },
  },
} satisfies Meta<typeof LeftLabeledField>;

export default meta;
type Story = StoryObj<typeof LeftLabeledField>;

const labelId = 'labelId';
const options: SelectOption<string>[] = [
  { label: 'Option 1', value: '1' },
  { label: 'Option 2', value: '2' },
];

export const Docs: Story = {
  args: {
    label: (
      <label style={{ background: '#ccc', padding: '8px' }} htmlFor={labelId}>
        Label text...
      </label>
    ),
    field: <Select id={labelId} searchable options={options} />,
  },
};
