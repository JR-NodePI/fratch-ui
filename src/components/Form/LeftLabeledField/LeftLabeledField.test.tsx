import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import LeftLabeledField from './LeftLabeledField';
import { type LeftLabeledFieldProps } from './LeftLabeledFieldProps';

describe('LeftLabeledField', () => {
  const initialProps: LeftLabeledFieldProps = {
    field: <input type="text" />,
    label: <label>mock-label</label>,
  };

  const getComponent = (props: LeftLabeledFieldProps = initialProps) => (
    <LeftLabeledField {...props} />
  );

  const setup = (props = initialProps) => render(getComponent(props));

  it('should render properly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should render with className', () => {
    const { container } = setup({
      ...initialProps,
      className: 'mock-class',
    });
    expect(container).toMatchSnapshot();
  });
});
