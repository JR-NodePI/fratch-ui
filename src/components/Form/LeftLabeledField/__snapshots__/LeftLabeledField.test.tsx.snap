// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LeftLabeledField > should render properly 1`] = `
<div>
  <div
    class="_wrapper_fca0d0"
  >
    <div
      class="_label_container_fca0d0"
    >
      <label>
        mock-label
      </label>
    </div>
    <div
      class="_field_container_fca0d0"
    >
      <input
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`LeftLabeledField > should render with className 1`] = `
<div>
  <div
    class="_wrapper_fca0d0 mock-class"
  >
    <div
      class="_label_container_fca0d0"
    >
      <label>
        mock-label
      </label>
    </div>
    <div
      class="_field_container_fca0d0"
    >
      <input
        type="text"
      />
    </div>
  </div>
</div>
`;
